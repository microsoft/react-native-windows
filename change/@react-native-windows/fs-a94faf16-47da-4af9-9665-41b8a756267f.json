{
  "type": "major",
  "comment": "Use '@react-native-windows/fs'",
  "packageName": "@react-native-windows/fs",
  "email": "ngerlem@microsoft.com",
  "dependentChangeType": "patch"
}
