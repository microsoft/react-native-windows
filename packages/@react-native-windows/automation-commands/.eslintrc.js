module.exports = {
    extends: ['@rnw-scripts'],
    parserOptions: {tsconfigRootDir : __dirname},
};
