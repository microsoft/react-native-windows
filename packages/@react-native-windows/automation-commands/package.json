{
  "name": "@react-native-windows/automation-commands",
  "version": "0.1.31",
  "description": "Allows controlling your react-native-windows application",
  "main": "lib-commonjs/index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git@github.com:microsoft/react-native-windows.git",
    "directory": "packages/@react-native-windows/automation-commands"
  },
  "private": true,
  "scripts": {
    "build": "rnw-scripts build",
    "clean": "rnw-scripts clean",
    "lint": "rnw-scripts lint",
    "lint:fix": "rnw-scripts lint:fix",
    "watch": "rnw-scripts watch"
  },
  "dependencies": {
    "@react-native-windows/automation-channel": "^0.5.11",
    "@typescript-eslint/eslint-plugin": "^5.30.5",
    "@typescript-eslint/parser": "^5.30.5"
  },
  "devDependencies": {
    "@jest/types": "^29.2.1",
    "@rnw-scripts/eslint-config": "1.1.14",
    "@rnw-scripts/just-task": "2.3.3",
    "@rnw-scripts/ts-config": "2.0.2",
    "@types/jest": "^29.2.2",
    "@types/node": "^14.14.22",
    "eslint": "^8.19.0",
    "prettier": "^2.4.1",
    "typescript": "^4.9.5"
  },
  "files": [
    "lib-commonjs",
    "README.md"
  ],
  "engines": {
    "node": ">= 14"
  }
}