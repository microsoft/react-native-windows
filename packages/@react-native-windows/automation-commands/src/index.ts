/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import dumpVisualTree from './dumpVisualTree';

export {dumpVisualTree};
