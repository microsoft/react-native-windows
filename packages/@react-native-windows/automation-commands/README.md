# @react-native-windows/automation-commands

`@react-native-windows/automation-commands` provides built-in commands that may
be used with `@react-native-windows/automation-channel` to make it easier to
test your application.

**This package is a work in progress**

## Adding to your application

**TBD**

## Commands

### dumpVisualTree

The `dumpVisualTree` command creates a JSON object corresponding to the XAML
tree under a given testID. This can be used in conjunction with snapshot
testing to validate that your native UI looks how you could expect.

```ts
import {dumpVisualTree} from '@react-native-windows/automation-commands';

test('Widget', async () => {
  const dump = await dumpVisualTree('widget-test-id');
  expect(dump).toMatchSnapshot();
});
```

#### Options
**TBD**