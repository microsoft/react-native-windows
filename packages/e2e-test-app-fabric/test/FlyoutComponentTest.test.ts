import {dumpVisualTree} from '@react-native-windows/automation-commands';
import {goToComponentExample} from './RNTesterNavigation';

beforeAll(async () => {
  await goToComponentExample('Flyout');
});

describe('FlyoutTests', () => {
  test('Flyout', async () => {
    const dump = await dumpVisualTree('flyout');
    expect(dump).toMatchSnapshot();
  });
});