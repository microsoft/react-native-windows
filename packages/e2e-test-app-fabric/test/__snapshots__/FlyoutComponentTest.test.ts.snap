// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlyoutTests Flyout 1`] = `
{
  "AutomationId": "flyout",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 562,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 31,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 800,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": [
    {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 59,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": [
        {
          "Clip": null,
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 39,
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "10,10,10,10",
          "Text": "Placement Options: ",
          "Top": 20,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 141,
          "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
        },
        {
          "Background": "#B3FFFFFF",
          "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
          "BorderThickness": "1,1,1,1",
          "Clip": null,
          "CornerRadius": "4,4,4,4",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 35,
          "HorizontalAlignment": "Left",
          "Left": 141,
          "Margin": "0,0,0,0",
          "Padding": "12,5,0,7",
          "Text": "",
          "Top": 20,
          "VerticalAlignment": "Top",
          "Visibility": "Visible",
          "Width": 200,
          "XamlType": "ReactNativePicker.ReactPickerView",
          "children": [
            {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Name": "LayoutRoot",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": [
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "#E4000000",
                  "BorderThickness": "2,2,2,2",
                  "Clip": null,
                  "CornerRadius": "7,7,7,7",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "-4,-4,-4,-4",
                  "Name": "HighlightBackground",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                },
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "4,4,4,4",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "Background",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                },
                {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 16,
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "1,0,0,0",
                  "Name": "Pill",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 3,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "12,5,0,7",
                  "Name": "ContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                  "children": [
                    {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Foreground": "#FF000000",
                      "HorizontalAlignment": "Left",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Padding": "0,0,0,0",
                      "Text": "top",
                      "Top": 0,
                      "VerticalAlignment": "Top",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    },
                  ],
                },
                {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#9E000000",
                  "Height": 12,
                  "HorizontalAlignment": "Right",
                  "Left": 0,
                  "Margin": "0,0,14,0",
                  "Name": "DropDownGlyph",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "Width": 12,
                  "XamlType": "Windows.UI.Xaml.Controls.IconElement",
                  "children": [
                    {
                      "Background": "#00000000",
                      "BorderBrush": null,
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Padding": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.Grid",
                      "children": [],
                    },
                  ],
                },
                {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#99000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "DescriptionPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "Popup",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Primitives.Popup",
                },
              ],
            },
          ],
        },
      ],
    },
    {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 78,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 59,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 200,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": [
        {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 38,
          "HorizontalAlignment": "Stretch",
          "Left": 20,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 20,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 160,
          "XamlType": "Microsoft.ReactNative.ViewControl",
          "children": [
            {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#FF000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
              "children": [
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1.5",
                  "Clip": null,
                  "CornerRadius": "3,3,3,3",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": [
                    {
                      "Background": null,
                      "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                      "BorderThickness": "1,1,1,1.5",
                      "Clip": null,
                      "CornerRadius": "3,3,3,3",
                      "FlowDirection": "LeftToRight",
                      "Height": 38,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 160,
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "children": [
                        {
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 35,
                          "HorizontalAlignment": "Stretch",
                          "Left": 1,
                          "Margin": "0,0,0,0",
                          "Top": 1,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 158,
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "children": [
                            {
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Height": 35,
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 158,
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "children": [
                                {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "Foreground": "#E4000000",
                                  "Height": 19,
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 8,
                                  "Margin": "0,0,0,0",
                                  "Padding": "0,0,0,0",
                                  "Text": "Open Flyout",
                                  "Top": 8,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "Width": 142,
                                  "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 96,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 137,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 200,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": [
        {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 56,
          "HorizontalAlignment": "Stretch",
          "Left": 20,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 20,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 160,
          "XamlType": "Microsoft.ReactNative.ViewControl",
          "children": [
            {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#FF000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
              "children": [
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1.5",
                  "Clip": null,
                  "CornerRadius": "3,3,3,3",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": [
                    {
                      "Background": null,
                      "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                      "BorderThickness": "1,1,1,1.5",
                      "Clip": null,
                      "CornerRadius": "3,3,3,3",
                      "FlowDirection": "LeftToRight",
                      "Height": 56,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 160,
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "children": [
                        {
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 54,
                          "HorizontalAlignment": "Stretch",
                          "Left": 1,
                          "Margin": "0,0,0,0",
                          "Top": 1,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 158,
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "children": [
                            {
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Height": 54,
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 158,
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "children": [
                                {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "Foreground": "#E4000000",
                                  "Height": 38,
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 8,
                                  "Margin": "0,0,0,0",
                                  "Padding": "0,0,0,0",
                                  "Text": "Open Flyout without Target",
                                  "Top": 8,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "Width": 142,
                                  "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 97,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 233,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 200,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": [
        {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 57,
          "HorizontalAlignment": "Stretch",
          "Left": 20,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 20,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 160,
          "XamlType": "Microsoft.ReactNative.ViewControl",
          "children": [
            {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#FF000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
              "children": [
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1.5",
                  "Clip": null,
                  "CornerRadius": "3,3,3,3",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": [
                    {
                      "Background": null,
                      "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                      "BorderThickness": "1,1,1,1.5",
                      "Clip": null,
                      "CornerRadius": "3,3,3,3",
                      "FlowDirection": "LeftToRight",
                      "Height": 57,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 160,
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "children": [
                        {
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 54,
                          "HorizontalAlignment": "Stretch",
                          "Left": 1,
                          "Margin": "0,0,0,0",
                          "Top": 1,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 158,
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "children": [
                            {
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Height": 54,
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 158,
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "children": [
                                {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "Foreground": "#E4000000",
                                  "Height": 38,
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 8,
                                  "Margin": "0,0,0,0",
                                  "Padding": "0,0,0,0",
                                  "Text": "Open Flyout with Offset",
                                  "Top": 8,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "Width": 142,
                                  "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 232,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 330,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": [
        {
          "Clip": null,
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 32,
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "10,10,10,10",
          "Text": "Text Input to Anchor flyout to: ",
          "Top": 200,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 300,
          "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
        },
        {
          "Background": "#B3FFFFFF",
          "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
          "BorderThickness": "1,1,1,1",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#E4000000",
          "Height": 32,
          "HorizontalAlignment": "Stretch",
          "Left": 300,
          "Margin": "0,0,0,0",
          "Padding": "10,5,6,6",
          "Text": "",
          "Top": 200,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 250,
          "XamlType": "Windows.UI.Xaml.Controls.TextBox",
          "children": [
            {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": [
                {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "BorderElement",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                },
                {
                  "Background": "#00FFFFFF",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#E4000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "1,1,1,1",
                  "Name": "ContentElement",
                  "Padding": "10,5,6,6",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ScrollViewer",
                  "children": [
                    {
                      "Background": null,
                      "BorderBrush": "#00FFFFFF",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "Root",
                      "Padding": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.Border",
                      "children": [
                        {
                          "Background": "#00FFFFFF",
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.Grid",
                          "children": [
                            {
                              "Background": null,
                              "BorderBrush": null,
                              "BorderThickness": "0,0,0,0",
                              "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Foreground": "#E4000000",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "10,5,6,6",
                              "Name": "ScrollContentPresenter",
                              "Padding": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.Controls.ScrollContentPresenter",
                              "children": [
                                {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 0,
                                  "Margin": "0,0,0,0",
                                  "Top": 0,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "XamlType": "Windows.UI.Xaml.FrameworkElement",
                                },
                              ],
                            },
                            {
                              "Background": null,
                              "BorderBrush": null,
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Padding": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.Controls.Grid",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#99000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "DescriptionPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;
